Sources/MapboxDirections/RouteResponse.swift

```diff
@@ -17,18 +17,26 @@ extension RouteResponse: Codable {
         case uuid
         case routes
         case waypoints
+        
+        // Matching-service specific keys. Used in decode only.
+        case matches = "matchings"
+        case tracepoints
     }
     
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        // Is this an edge-case, where we are creating a route response from the map matching service?
+        let fromMatchingService = (decoder as? DirectionsDecoder)?.decodingRouteResponseFromMatchService ?? false
```

Comment:
The former MapMatchingResponse and RouteResponse functionality should not be unified into a single class. Doing so results in special cases within the implementation, which could turn into spaghetti code in the future as we add support for the Optimization API (#107) and Matrix API.

https://github.com/mapbox/mapbox-directions-swift/issues/391#issue-527602027 proposed:

> For the case that MapMatchingResponse is meant to handle, it would be better for MapMatchingResponse to produce Match objects; `calculateRoutes(matching:completionHandler:)` could call a new initializer on RouteResponse that takes a MapMatchingResponse as input.

What would be the downside of first interpreting a Map Matching API response as a series of Matches in a MapMatchingResponse, then converting the Matches to Routes in a RouteResponse? Could this approach lead to better handling of submatches than #386?